Dockerfile

```diff
@@ -26,22 +30,26 @@ ADD ./cmd/ $APP_ROOT/src/cmd/
 ADD ./pkg/ $APP_ROOT/src/pkg/
 
 ARG SERVER_LDFLAGS
+ARG IMAGE_VARIANT
+# Build server for deployment
 RUN go build -ldflags "${SERVER_LDFLAGS}" ./cmd/rekor-server
-RUN CGO_ENABLED=0 go build -gcflags "all=-N -l" -ldflags "${SERVER_LDFLAGS}" -o rekor-server_debug ./cmd/rekor-server
-
-# Multi-Stage production build
-FROM golang:1.24.2@sha256:18a1f2d1e1d3c49f27c904e9182375169615c65852ace724987929b910195b2c AS deploy
+# Build server for debugger
+RUN if [ "${IMAGE_VARIANT}" = "debug" ]; then CGO_ENABLED=0 go build -gcflags "all=-N -l" -ldflags "${SERVER_LDFLAGS}" -o rekor-server_debug ./cmd/rekor-server; fi
 
+# Multi-stage deployment build
+FROM gcr.io/distroless/static-debian12:nonroot@sha256:c0f429e16b13e583da7e5a6ec20dd656d325d88e6819cafe0adb0828976529dc AS deploy
 # Retrieve the binary from the previous stage
 COPY --from=builder /opt/app-root/src/rekor-server /usr/local/bin/rekor-server
 
+# Multi-stage debugger build
+FROM golang:1.24.2@sha256:18a1f2d1e1d3c49f27c904e9182375169615c65852ace724987929b910195b2c AS debug
+# Copy debugger server
+COPY --from=builder /opt/app-root/src/rekor-server_debug /usr/local/bin/rekor-server
```

Comment:
I tried building this with

```
docker build . --build-arg IMAGE_VARIANT=deploy
```
and it failed at this step. It's missing the debug binary built on line 37. I'm not sure if the Dockerfile syntax is smart enough to only run certain steps depending on the build arguments.

If it's not easy to add conditionals in one dockerfile I'm fine with splitting it in two.

Reply:
Debugging this now, it looks to be due to building with CGO enabled, which will fail on distroless. I've got a fix, pushing it up now

Reply:
Eh, now the build times out. Lemme just switch base images.

Reply:
Oh I just reread your statement, yep I see the problem.